internal/datasource/config/config_writer.go

```diff
@@ -28,52 +31,56 @@ const (
 
 type (
 	ConfigWriterInterface interface {
-		Write(ctx context.Context, filesURL, tenantID, instanceID string) (skippedFiles CacheContent, err error)
+		Write(ctx context.Context, request *v1.ManagementPlaneRequest_ConfigApplyRequest,
+			instanceID string) (skippedFiles CacheContent, err error)
 		Complete(ctx context.Context) (err error)
-		SetConfigClient(configClient client.ConfigClientInterface)
-		Rollback(ctx context.Context, skippedFiles CacheContent, filesURL, tenantID, instanceID string) error
+		SetConfigClient(configClient client.ConfigClient)
+		Rollback(ctx context.Context, skippedFiles CacheContent,
+			request *v1.ManagementPlaneRequest_ConfigApplyRequest, instanceID string) error
 	}
 
 	ConfigWriter struct {
-		configClient       client.ConfigClientInterface
+		configClient       client.ConfigClient
```

Comment:
if any of these aren't an interface, should they be pointers?

Reply:
I don't think so but I could be wrong, configClient and fileCache are interfaces and currentFileCache is CacheContent which is an alias for ` map[string]*v1.FileMeta`